File: AutomaticTimetableGeneratorSystem-main/timetable-frontend/src/services/departmentService.js
import api from "./api";

const BASE_URL = "/departments";

const departmentService = {

    getAllDepartments: async () => {

        const response = await api.get(BASE_URL);

        return response.data;

    },

    getDepartmentById: async (id) => {

        const response = await api.get(`${BASE_URL}/${id}`);

        return response.data;

    },

    createDepartment: async (department) => {

        const response = await api.post(

            BASE_URL,

            department

        );

        return response.data;

    },

    updateDepartment: async (id, department) => {

        const response = await api.put(

            `${BASE_URL}/${id}`,

            department

        );

        return response.data;

    },

    deleteDepartment: async (id) => {

        const response = await api.delete(

            `${BASE_URL}/${id}`

        );

        return response.data;

    }

};

export default departmentService;

File: AutomaticTimetableGeneratorSystem-main/timetable-backend/src/main/java/com/timetable/service/DepartmentService.java
package com.timetable.service;

import java.util.List;

import com.timetable.dto.department.DepartmentRequest;
import com.timetable.dto.department.DepartmentResponse;

public interface DepartmentService {

    DepartmentResponse createDepartment(DepartmentRequest request);

    DepartmentResponse getDepartmentById(Long departmentId);

    DepartmentResponse getDepartmentByCode(String departmentCode);

    List<DepartmentResponse> getAllDepartments();

    DepartmentResponse updateDepartment(Long departmentId,
                                        DepartmentRequest request);

    void deleteDepartment(Long departmentId);

}